Update README.md

frederike-ramin opened the pull request.

Commit:
WIP: Update README.md

Commit:
Add prototype pictures

Commit:
Add pictures and configuration description to readme

GoesOnTangents commented on README.md:
```diff
@@ -1,16 +1,93 @@
+# EMPOWER
 
-First frontend of our app.
+An app that visualizes the effects of changing inputs on the results of given logistic regression outputs.
```
@justus-hildebrand

justus-hildebrand replied:
```suggestion
Empower is an app that shows the relations between changes in feature vectors and the resulting effects on predictions made by pre-trained logistic regressors. It does this by providing users with an input interface to make changes to feature values and on the fly presenting the resulting effects on the model's prediction results.
```

frederike-ramin replied:
That is sooo complicated..It might be accurate, bu probably good as second sentence? So like "visualization" first and then "To be more exact <your part>".

GoesOnTangents commented on README.md:
```diff
@@ -1,16 +1,93 @@
 
-First frontend of our app.
+An app that visualizes the effects of changing inputs on the results of given logistic regression outputs.
+It can be used for any kind of data. For the example configs, we used a health use case to predict tuberculosis, diabetes, asthma and COPD on certain parameters like age, weight, height and smoking behavior.
```
```suggestion
It can be used for any kind of clean data. For the example configuration, we use a health use case to predict tuberculosis, diabetes, asthma and COPD on life style choices and patient parameters like age, weight, etc.
```

GoesOnTangents commented on README.md:
```diff
@@ -1,16 +1,93 @@
+## Steps to run
+1. Clone the project
+2. Configure
+    * Connect to a running server of [patientEmpowerment](https://github.com/KBorchar/patientEmpowerment) via editing `assets/server.conf` and load `assets/feature-config.json` and `assets/models.json` from there. 
```
Are we migrating these to patientempowerment?

frederike-ramin replied:
Do you mean migrating the backend repo? I'd vote yes.

GoesOnTangents replied:
k then this needs a TODO

GoesOnTangents commented on README.md:
```diff
@@ -1,16 +1,93 @@
+## Steps to run
+1. Clone the project
+2. Configure
+    * Connect to a running server of [patientEmpowerment](https://github.com/KBorchar/patientEmpowerment) via editing `assets/server.conf` and load `assets/feature-config.json` and `assets/models.json` from there. 
```
```suggestion
    * Connect to a running server of [patientEmpowerment](https://github.com/KBorchar/patientEmpowerment) via editing `assets/server.conf`, then load `assets/feature-config.json` and `assets/models.json` from there. 
```

GoesOnTangents commented on README.md:
```diff
@@ -1,16 +1,93 @@
+2. Configure
+    * Connect to a running server of [patientEmpowerment](https://github.com/KBorchar/patientEmpowerment) via editing `assets/server.conf` and load `assets/feature-config.json` and `assets/models.json` from there. 
+    * OR:
+    * To configure the input and output manually, edit `assets/feature-config.json` and `assets/models.json` as described further down.
```
```suggestion
    * To configure the input and output manually, edit `assets/feature-config.json` and `assets/models.json` (see Configuration).
```

GoesOnTangents commented on README.md:
```diff
@@ -1,16 +1,93 @@
+    * Connect to a running server of [patientEmpowerment](https://github.com/KBorchar/patientEmpowerment) via editing `assets/server.conf` and load `assets/feature-config.json` and `assets/models.json` from there. 
+    * OR:
+    * To configure the input and output manually, edit `assets/feature-config.json` and `assets/models.json` as described further down.
+The example files were created with the [backend](https://github.com/KBorchar/patientEmpowerment).
```
```suggestion
The example files were created by the [backend](https://github.com/KBorchar/patientEmpowerment).
```

GoesOnTangents commented on README.md:
```diff
@@ -1,16 +1,93 @@
+    * OR:
+    * To configure the input and output manually, edit `assets/feature-config.json` and `assets/models.json` as described further down.
+The example files were created with the [backend](https://github.com/KBorchar/patientEmpowerment).
+3. To configure the displayed titles of the output, edit `assets/labels.conf`.
```
```suggestion
3. To configure the user-facing names of the output labels, edit `assets/labels.conf`.
```

GoesOnTangents commented on README.md:
````diff
@@ -1,16 +1,93 @@
+}
+```
+It represents the possible input fields.
+As key, you use the feature name, give it a title and decide whether it should be represented as a slider or radio choice buttons. For radio buttons, specify a list of available choices with their correlated number representation in your ML algorithm. For a slider, specify the minimal and maximal value that the user should be able to input.
````
```suggestion
As key, you use the feature name. Give it a user-facing title and decide whether it should be represented as a slider or choice. For choices, specify a list of available choices with their correlating value in the original data set. For a slider, specify the minimal and maximal value that the user should be able to input.
```

GoesOnTangents commented on README.md:
```diff
@@ -1,16 +1,93 @@
 
-A few resources to get you started if this is your first Flutter project:
+#### models.json
+In `models.json` you specify, per label, the ML determined coefficients for each feature as well as their means. It should look similar to this:
```
```suggestion
In `models.json` you specify, per label, the coefficients for each feature as well as their means in the original data set. It should look similar to this:
```

GoesOnTangents commented on README.md:
````diff
@@ -1,16 +1,93 @@
+  ...
+}
+```
+First key is the label name, in this case "COPD". inside the "feature" argument you then can specify a list of features that influence this label with the respective coefficients (at key "coeff") and means (at key "mean"). Repeat this for all features and labels.
````
```suggestion
First key is the label, in this case "asthma". "features" contains a list of all respective features, their coefficients, and their means.
```

GoesOnTangents replied:
```suggestion
First key is the label name, in this case "COPD". inside the "feature" argument you then can specify a list of features that influence this label with the respective coefficients (at key "coeff") and means (at key "mean"). Repeat this for all features and labels. This file is auto-generated, whenever the [backend] fits the corresponding model.
```

GoesOnTangents commented on README.md:
```diff
@@ -1,16 +1,93 @@
-- [Lab: Write your first Flutter app](https://flutter.dev/docs/get-started/codelab)
-- [Cookbook: Useful Flutter samples](https://flutter.dev/docs/cookbook)
+#### labels.conf
+In `assets/labels.conf` you specify all labels that your want to predict for, that also have a representation in `assets/models.json` (!) as a simple list with the key "labels".
```
```suggestion
`assets/labels.conf` specifies all labels to predict for, and their user-facing names.
```

GoesOnTangents commented on README.md:
```diff
@@ -1,16 +1,93 @@
+
+
+## How to use
+There are two visual representations of the same concept: Bars on startup and Bubbles once you swipe one page to the right.
```
```suggestion
There are two visual representations of the same concept: A bar chart (default), and a more interactive prototype. Swipe right on the bar chart screen to access it.
```

GoesOnTangents commented on README.md:
```diff
@@ -1,16 +1,93 @@
+
+## How to use
+There are two visual representations of the same concept: Bars on startup and Bubbles once you swipe one page to the right.
+Should you have left out some input parameters, the app will assume the means given in `assets/models.json`.
```
```suggestion
Should any inputs be empty, the app will assume the corresponding feature's mean (from `assets/models.json`).
```

GoesOnTangents commented on README.md:
```diff
@@ -1,16 +1,93 @@
+
+### Bars
+![Alt text](/assets/bars_prototype.png "Bars Prototype")
+On the left side, you can put in your data and press the button in the middle to get the output. You can also change the input while the output is displayed or hide the output bars again by pressing the same button in the middle. To reset to means, press the floating button on the bottom right.
```
```suggestion
On the left side, users can input their data and press the button in the middle to get the output. The output reacts in real time to any changes on the input. You can hide it again by pressing the same button in the middle. To reset all inputs to their means, press the floating button on the bottom right.
```

Commit:
Correct server.conf usage

GoesOnTangents commented on README.md:
```diff
@@ -1,16 +1,93 @@
+
+### Bubbles
+![Alt text](/assets/bubbles_prototype.png "Bubbles Prototype")
+In Bubbles Prototype the input fields have a bubble representation. They are located at the top of the screen. The output is also in a circular form in the screen center around a center image. Each output bubble representing one ML prediction label. You can input your data by dragging or clicking an input bubble. A dialog opens and asks you to input your data. By pressing "ok" the change is propa ...
```
```suggestion
In the 'Bubbles' prototype the input fields have a bubble representation. They are initially located at the top of the screen. The output is represented by fixed bubble, arranged around the center image. Each output bubble represents one label to predict on. Users can input their data by dragging or tapping an input bubble, upon which an input dialog opens. By pressing "ok" the change is propagated to the output bubbles and their size change according to the new probabilities. Additionally, there are particles, small bubbles, flying from the input to the influenced output bubble(s) to indicate their correlation.
```

frederike-ramin replied:
Just fixed some typos
```suggestion
In the 'Bubbles' prototype the input fields have a bubble representation. They are initially located at the top of the screen. The output is represented by fixed bubbles, arranged around the center image. Each output bubble represents one label to predict on. Users can input their data by dragging or tapping an input bubble, upon which an input dialog opens. By pressing "ok" the change is propagated to the output bubbles and their size changes according to the new probabilities. Additionally, there are particles, small bubbles, flying from the input to the influenced output bubble(s) to indicate their correlation.
```

GoesOnTangents commented on README.md:
```diff
@@ -1,16 +1,93 @@
+
+## Contributing
+We are happy if you'd want to contribute to the project. To get into the development, you might want to take some things into account.
+### Naming
```
honestly I would remove the whole # Naming section

frederike-ramin replied:
I just thought it might be confusin to people not familiar with ML language.

GoesOnTangents replied:
Then they shouldn't use this app :) <3

frederike-ramin replied:
Sure. :D

Commit:
Update README.md

Co-Authored-By: Tenta <kim-pascal.borchart@student.hpi.de>

Commit:
Update README.md

Co-Authored-By: Tenta <kim-pascal.borchart@student.hpi.de>

Commit:
Apply suggestions from code review

Co-Authored-By: Tenta <kim-pascal.borchart@student.hpi.de>

Commit:
Work in review comments

davidimmhahn commented on README.md:
```diff
@@ -1,16 +1,111 @@
+
+### Bars
+![Alt text](/assets/bars_prototype.png "Bars Prototype")
+On the left side, users can input their data and press the button in the middle to get the output. The output reacts in real time to any changes on the input. You can hide it again by pressing the same button in the middle. To reset all inputs to their means, press the floating button on the bottom right.
```
```suggestion
On the left side, users can input their data and press the button in the middle to get the output. The output reacts in real time to any changes on the input. You can hide it again by pressing the same button in the middle. To reset all inputs to their means, press the floating action button on the bottom right.
```

davidimmhahn commented on README.md:
```diff
@@ -1,16 +1,111 @@
+
+### Bubbles
+![Alt text](/assets/bubbles_prototype.png "Bubbles Prototype")
+In the 'Bubbles' prototype the input fields have a bubble representation. They are initially located at the top of the screen. The output is represented by fixed bubbles, arranged around the center image. Each output bubble represents one label to predict on. Users can input their data by dragging or tapping an input bubble, upon which an input dialog opens. By pressing "ok" the change is propaga ...
```
```suggestion
In the 'Bubbles' prototype the input fields have a bubble representation. They are initially located at the top of the screen. The output is represented by fixed label bubbles, arranged around the center image. Each label bubble represents one label to predict on. Users can input their data by dragging or tapping an input bubble, upon which an input dialog opens. By pressing "ok" the change is propagated to the label bubbles and their size changes according to the new probabilities. Additionally, there are particles, small bubbles, flying from the input to the influenced label bubble(s) to indicate their correlation.
```

davidimmhahn commented on README.md:
```diff
@@ -1,16 +1,111 @@
+## Steps to run
+1. Clone the project
+2. Configure
+    * Connect to a running server of [patientEmpowerment](https://github.com/KBorchar/patientEmpowerment) via editing `assets/server.conf`, then load `assets/feature-config.json` and `assets/models.json` from there. 
```
change URL to code in organization repo once migrated

davidimmhahn replied:
```suggestion
    * Connect to a running server of [patientEmpowerment](https://github.com/patientempowerment/patientEmpowerment) via editing `assets/server.conf`, then load `assets/feature-config.json` and `assets/models.json` from there. 
```

davidimmhahn reviewed (commented):
nice work!

justus-hildebrand committed:
Apply suggestions from code review

Co-Authored-By: David Hahn <davidimmhahn@users.noreply.github.com>

davidimmhahn committed:
Update README.md

Even though this sentence is super complicated. We are going with it until someone has a better version of it :)

Co-Authored-By: Justus Hildebrand <justus.hildebrand@student.hpi.de>
